### DISCUSSION TITLE
Cool externals to be considered as addition to PlugData plugin version

### DISCUSSION THREAD

**Discussion by alfonso73** [plugdata-team/plugdata#129] [General]
I'm just testing FluCoMa...and it's just great! https://github.com/flucoma/flucoma-pd
Onset detection (both on buffers and on live input) and various other analysis features. Just great!

**Comment by timothyschoen** [comment]
I'm gonna try that for sure! I can already think of so many applications for onset detection in the DAW: I've been wanting to make a chorus effect that ducks the chorused signal on transients to keep them punchy!

It's sad that externals can't work normally in the DAW, I don't want to add too many externals, but this looks very nice.

**Reply by jayrope** [reply]
Level-based transient detection isn't too difficult in Vanilla already. Run audio through an env~, smooth out with two different line~settings (experiment) and have a bang sent, when the faster one exceeds the slower one. I wrote myself a more luxurious abstraction already decades ago and never needed more (yet, i guess).

**Comment by alfonso73** [comment]
I think that a good set of onset detection stuff, FFT stuff and a good set of bandlimited oscillators would be enough to fully explore the power of pure data as a plugin (obviously as additions to the wonderful and powerful ELSE stuff!)

**Comment by alfonso73** [comment]
@timothyschoen look at the great FluCoMa based work by Rodrigo Constanzo https://github.com/rconstanzo/SP-tools 
With FluCoMa in PlugData a whole world would be disclosed

**Reply by timothyschoen** [reply]
I'm still a bit unsure about adding more externals into PlugData. I feel like getting externals to work in the plugin would be a much better option for the future. If that really ends up not working, I'll consider it. I'm also thinking about writing a little tutorial for adding your own externals to PlugData, which is actually doable.

The reason I'm hesitant about this is that the more externals I include, the more I will have to maintain in the future, which becomes problematic once projects start to be abandoned by their original maintainers. You can see that this already affects the pd-extended/purr-data project. I would be interested in adding some onset detection and pitch detection stuff into ELSE eventually, if porres is interested in that as well.

**Comment by michaelmariaott** [comment]
Has anyone been able to successfully compile PlugDate with Flucoma included to bea able to use Flucoma externals in plugins exported from PlugData?

**Reply by dromer** [reply]
What do you mean with "plugins exported from plugdata"? If you mean using the Heavy Compiler then this will most certainly not work.
